discovery-infra/test_infra/utils.py

```diff
@@ -801,13 +808,57 @@ def update_hosts(client, cluster_id, libvirt_nodes, update_hostnames=False, upda
 
     client.update_hosts(cluster_id=cluster_id, hosts_with_roles=roles, hosts_names=hostnames)
 
+
 def get_assisted_controller_status(kubeconfig):
     log.info("Getting controller status")
-    command = f"oc --insecure-skip-tls-verify --kubeconfig={kubeconfig} --no-headers=true -n assisted-installer get pods -l job-name=assisted-installer-controller"
+    command = f"oc --insecure-skip-tls-verify --kubeconfig={kubeconfig} --no-headers=true -n assisted-installer " \
+              f"get pods -l job-name=assisted-installer-controller"
     response = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
     if response.returncode != 0:
         log.error(f'failed to get controller status: {response.stderr}')
         return b''
 
     log.info(f'{response.stdout}')
     return response.stdout
+
+
+def get_network_interface_ip(interface):
+    if len(interface.ipv4_addresses) > 0:
+        return interface.ipv4_addresses[0].split("/")[0]
+    if len(interface.ipv6_addresses) > 0:
+        return interface.ipv6_addresses[0].split("/")[0]
+    return " "
+
+
+def get_inventory_host_nics_data(host: dict):
+    inventory = models.Inventory(**json.loads(host["inventory"]))
+    interfaces_list = [models.Interface(**interface) for interface in inventory.interfaces]
+    return [{'name': interface.name, 'model': interface.product, 'mac': interface.mac_address,
+             'ip': get_network_interface_ip(interface), 'speed': interface.speed_mbps} for interface in interfaces_list]
+
+
+def get_inventory_host_ips_data(host: dict):
+    nics = get_inventory_host_nics_data(host)
+    return [nic["ip"] for nic in nics]
+
+
+def get_inventory_host_reachable_ips(host: dict, port: int):
+    ips = get_inventory_host_ips_data(host)
+    return [ip for ip in ips if connection.is_open(ip, port)]
+
+
+def get_masters_from_cluster(cluster: dict):
+    return [host for host in cluster["hosts"] if host["role"] == consts.NodeRoles.MASTER]
+
+
+def get_api_vip_from_cluster(cluster: dict):
+    if cluster.get("api_vip"):
+        logging.info("API VIP is set: %s", cluster["api_vip"])
+        return cluster["api_vip"]
+    # workaround for MGMT-3583
+    if cluster.get("user_managed_networking") or cluster.get("user-managed-networking"):
+        logging.info("API VIP is not set, taking masters reachable ip")
+        masters = get_masters_from_cluster(cluster)
+        ips = get_inventory_host_reachable_ips(masters[0], 443)
+        logging.info("master %s first reachable ip is %s", masters[0], ips[0])
+        return ips[0]
```

Comment:
I'm not sure this is a valid way to get the API VIP.
You must search for 'api' interface by name. Other than that, you're right it could be any master.

Reply:
how u know which interface is api vip? If u have an idea will be glad to implement

Reply:
It has the name `api` since that's how it is being generated.
i.e.
```
4: api@ens3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default 
    link/ether 00:1a:4a:54:c6:3d brd ff:ff:ff:ff:ff:ff
5: ingress@ens3: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP mode DEFAULT group default 
    link/ether 00:1a:4a:46:45:d8 brd ff:ff:ff:ff:ff:ff
```

Reply:
u don't have api vip in None platform